bin/python/ripple/build/module/Default.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals)
+
+import os
+
+from beast.build.Module import Module
+
+from beast.build.Function import Env, compose, for_tags, not_tags
+
+Append = Env.Append
+Prepend = Env.Prepend
+
+
+MODULE = Module(
+    files=compose(
+        for_tags(
+            'debug',
+            Append(CPPDEFINES=['DEBUG', '_DEBUG']),
+        ),
+
```

Comment:
@rec: I really like the interface you have here. Very nice. However, I don't like the complexity needed to support this interface. I can't do better, but I'm also concerned about anyone on the team other than you supporting this python code. Looking at the implementation, there are lots of functions passed around. It makes the code hard to follow for someone (like me) who isn't used to the style. For example:

1) Look at what `for_tags` does. This is defined in `Function.py`

2) That calls: `_tagger` with the items and a matcher.

3) `_tagger` returns a function that iterates through all the callable objects that pass the matcher with the non-callable objects (I did a double-take on `(functions.append if callable(i) else tags.add)(i)`)

4) `Append` is `Env.Append`. Env is defined in `Function.py` as `EnvClass`. `EnvClass` doesn't have an attribute `Append`, but it does define a `__getattr__` function that return a function that returns a function (two levels) that calls a function on the build env.

5) `compose` is defined in `Function.py` and returns a function that calls all of the arguments originally passed to `compose`.

6) The `Module` `__init__` function looks at the `files` parameter. If it's callable, it saves it, if not there are other special cases.

I'm also having some problems navigating all the python files that support our build system.

I don't know how actionable that feedback is, but I did want to share the concern.